mnemosyne/core/hygiene.py

```
@@ -0,0 +1,543 @@
+    try:
+        _ensure_hygiene_log_table(conn)
+        cursor = conn.cursor()
+        now = datetime.now().isoformat()
+
+        for c in candidates:
+            effective_action = action if action != "keep" else c.suggested_action
+
+            try:
+                # Fetch original content + metadata for audit log
+                cursor.execute(
+                    f"SELECT content, metadata_json FROM {c.table_name} WHERE id = ?",
```

Ruff (0.15.20)

[error] Possible SQL injection vector through string-based query construction

(S608)